ehrapy/_utils_data.py

```diff
@@ -0,0 +1,53 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any
+
+import numpy as np
+
+if TYPE_CHECKING:
+    from anndata import AnnData
+
+
+def are_dataset_equal(arr1: np.ndarray, arr2: np.ndarray) -> np.bool_:
+    """Check if two arrays are equal member-wise.
+
+    Note: Two NaN are considered equal.
+
+    Args:
+        arr1: First array to compare
+        arr2: Second array to compare
+
+    Returns:
+        True if the two arrays are equal member-wise
+    """
+    return np.all(np.equal(arr1, arr2, dtype=object) | ((arr1 != arr1) & (arr2 != arr2)))
+
+
+def to_missing_data_matrix(data: np.ndarray) -> np.ndarray[Any, np.dtype[np.bool_]]:  # pragma: no cover
+    """Check if a AnnData value is missing.
+
+    Args:
+        data: The AnnData data to check
```

Comment:
Let's not call this data but rather `matrix` or so maybe? In the AnnData jargon we are always talking about matrices that are in X or layers.